src/pharmpy/tools/run.py

```diff
@@ -857,22 +856,82 @@ def _get_delta(model):
         return df.sort_values(by=[f'd{rank_type_name}'], ascending=False)
 
 
-def _is_valid_res(res, errors_allowed):
-    if not res or np.isnan(res.ofv):
+class ArrayEvaluator:
+    def __init__(self, x):
+        self.x = x
+
+    def __lt__(self, value):
+        return all(e < value for e in self.x)
+
+    def __eq__(self, value):
+        return all(e == value for e in self.x)
+
+    def __le__(self, value):
+        return all(e <= value for e in self.x)
+
+    def __ge__(self, value):
+        return all(e >= value for e in self.x)
+
+    def __gt__(self, value):
+        return all(e > value for e in self.x)
+
+
+def _strictness_filtering(res: ModelfitResults, statement: str) -> bool:
```

Comment:
I think it could be good to have unit tests on this function even though it's a private function.